### ISSUE TITLE
Rename results_publishing

### ISSUE THREAD

**Issue by IanHoang** [opensearch-project/opensearch-benchmark#311]
**Is your feature request related to a problem? Please describe.**

Results_publishing is kinda clunky. Let's update this to publishing

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by AManaswini** [comment]
Hi, is this the results_publisher.py in the osbenchmark folder.I don't find anything named results_publishing.py

**Comment by IanHoang** [comment]
Yes, @AManaswini this is the results_publisher.py. `results_publishing` is in the `benchmark.ini` template and requires changing as well. You may need to re-revert this commit or you can reference the changes that were partially made in this commit. https://github.com/opensearch-project/opensearch-benchmark/pull/353

**Comment by soumyatanwar** [comment]
@IanHoang , @AManaswini - a follow up question here, when I look up "results_publishing" across the repository it also brings up code references which partially include this search term.  `configure_results_publishing_params` in `benchmark.py` and `include_in_results_publishing` in `workload.py`. 

Should these also be renamed to be consistent with the refactoring?
